addons/mrp/models/mrp_production.py

```diff
@@ -1581,6 +1605,21 @@ def button_mark_done(self):
                     'res_id': self.id,
                     'target': 'main',
                 }
+            if self.user_has_groups('mrp.group_mrp_reception_report') and self.picking_type_id.auto_show_reception_report:
+                lines = self.move_finished_ids.filtered(lambda m: m.product_id.type == 'product' and m.state != 'cancel' and m.quantity_done and not m.move_dest_ids)
+                if lines:
+                    # don't show reception report if all already assigned/nothing to assign
+                    wh_location_ids = self.env['stock.location'].search([('id', 'child_of', self.picking_type_id.warehouse_id.view_location_id.id), ('usage', '!=', 'supplier')]).ids
+                    if self.env['stock.move'].search([
+                            ('state', 'in', ['confirmed', 'partially_available', 'waiting', 'assigned']),
+                            ('product_qty', '>', 0),
+                            ('location_id', 'in', wh_location_ids),
+                            ('move_orig_ids', '=', False),
+                            ('raw_material_production_id', 'not in', self.ids),
+                            ('product_id', 'in', lines.product_id.ids)], limit=1):
```

Comment:
You can't check show_allocation instead ?

Reply:
Good point. Changed it (also discovered my _compute_show_allocation was wrong and fixed it)